metrics/group_workload_keys.go

```diff
@@ -26,5 +29,9 @@ func NewWorkloadKeysMetricGroup(info *MetricInfo) *WorkloadKeysMetricGroup {
 
 func (w *WorkloadKeysMetricGroup) GetMetrics(status *models.FullStatus) {
 	scope := w.GetScopeOrExit("default")
+	if status == nil || status.Cluster == nil || status.Cluster.Workload == nil || status.Cluster.Workload.Keys == nil || &status.Cluster.Workload.Keys.Read == nil {
```

Comment:
nit: this can be moved to a helper method `IsValidStatus()`

Reply:
I will move the high level ones in the next pr, whole sections can be missing here.